src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -285,9 +285,8 @@ public void addInnerHits(String name, InnerHitsContext.BaseInnerHits context) {
         innerHitsContext.addInnerHitDefinition(name, context);
     }
 
-    @Nullable
-    public Query parseInnerQuery() throws IOException, QueryParsingException {
-        // move to START object
+    public QueryBuilder toQueryBuilder() throws IOException {
```

Comment:
Nested is confusing cause we have a nested query and filter. Inner is better but taken. I can live with the QueryBuilder temporarily and call it parseInnerQueryBuilder for now I guess.